src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryArgumentSetter.java

```diff
@@ -0,0 +1,111 @@
+ * as pipeline arguments
+ */
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name(BigQueryArgumentSetter.NAME)
+@Description("Argument setter for dynamically configuring pipeline from BiqQuery table.")
+public final class BigQueryArgumentSetter extends Action {
+
+  private static final Logger LOG = LoggerFactory.getLogger(BigQueryArgumentSetter.class);
+  public static final String NAME = "BigQueryArgumentSetter";
+  private BigQueryArgumentSetterConfig config;
+
+  @Override
+  public void configurePipeline(PipelineConfigurer configurer) {
+    super.configurePipeline(configurer);
+    StageConfigurer stageConfigurer = configurer.getStageConfigurer();
+    FailureCollector collector = stageConfigurer.getFailureCollector();
+    config.validate(collector);
+  }
+
+  @Override
+  public void run(ActionContext context) throws Exception {
+    config.validate(context.getFailureCollector());
+
+    QueryJobConfiguration queryConfig = config.getQueryJobConfiguration();
+    JobId jobId = JobId.newBuilder().setRandomJob().build();
+
+    // API request - starts the query.
+    Credentials credentials = config.getServiceAccountFilePath() == null ?
+      null : GCPUtils.loadServiceAccountCredentials(config.getServiceAccountFilePath());
+    BigQuery bigQuery = GCPUtils.getBigQuery(config.getProject(), credentials);
+    Job queryJob = bigQuery.create(JobInfo.newBuilder(queryConfig).setJobId(jobId).build());
+
+    LOG.info("Executing SQL as job {}.", jobId.getJob());
+    LOG.debug("The BigQuery SQL  {}", queryConfig.getQuery());
+
+    // Wait for the query to complete
+    queryJob.waitFor();
+
+    // Check for errors
+    if (queryJob.getStatus().getError() != null) {
+      throw new RuntimeException(queryJob.getStatus().getExecutionErrors().toString());
+    }
+
+    TableResult queryResults = queryJob.getQueryResults();
+    if (queryResults.getTotalRows() == 0 || queryResults.getTotalRows() > 1) {
+      throw new RuntimeException(String.format("The query result total rows should be \"1\" but is \"%d\"",
+                                               queryResults.getTotalRows()));
+    }
+
+    Schema schema = queryResults.getSchema();
+    FieldValueList row = queryResults.iterateAll().iterator().next();
+
+    for (int i = 0; i < schema.getFields().size(); i++) {
+      Field field = schema.getFields().get(i);
+      String name = field.getName();
+      FieldValue fieldValue = row.get(name);
+      FieldValue.Attribute attribute = fieldValue.getAttribute();
+      if (attribute != FieldValue.Attribute.PRIMITIVE) {
+        context.getFailureCollector().addFailure(String.format("Field %s is not primitive type.", field.getName()),
+                                                 "Only primitive types are allowed.");
+        context.getFailureCollector().getOrThrowException();
+      }
+      context.getArguments().set(name, fieldValue.getStringValue());
```

Comment:
Primitive type also includes `Bytes` and `TIMESTAMP` types. what is the string value for those type? Also I dont think we should allow bytes type

Reply:
Type ``Bytes`` is removed.
Type ``TIMESTAMP`` string value is seconds since epoch `` (e.g. 1408452095.22 == 2014-08-19 07:41:35.220 -05:00)``

Reply:
can you add comment about timestamp type?